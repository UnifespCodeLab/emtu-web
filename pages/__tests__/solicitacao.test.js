import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'

import ReportPage from '~/pages/solicitacao'

describe('Pages / ReportPage', () => {
  const localVue = createLocalVue()
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()

    wrapper = mount(ReportPage, {
      localVue,
      vuetify,
      stubs: {
        'router-link': RouterLinkStub,
      },
    })
  })

  it('should render the page', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render correct elements', () => {
    const alert = wrapper.findComponent({ name: 'v-alert' })
    expect(alert.props('value')).toBe(false)

    const homeButton = wrapper.findAllComponents({ name: 'v-btn' }).at(1)
    expect(homeButton.text()).toBe('enviar')
  })

  describe('when user clicks in submit', () => {
    beforeEach(() => {
      const submitButton = wrapper.findAllComponents({ name: 'v-btn' }).at(1)
      submitButton.trigger('click')
    })

    it('should render updated page', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('should render correct elements', () => {
      const alert = wrapper.findComponent({ name: 'v-alert' })
      const homeButton = wrapper.findAllComponents({ name: 'v-btn' }).at(1)

      expect(alert.props('value')).toBe(true)
      expect(homeButton.text()).toBe('voltar para home')
    })
  })
})
