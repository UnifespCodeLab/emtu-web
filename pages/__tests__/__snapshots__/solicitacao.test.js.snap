// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / ReportPage should render the page 1`] = `
<div
  class="report-page"
>
  <transition-stub
    class="report-page__alert"
    name="scale-transition"
  >
    <div
      class="v-alert v-sheet theme--dark error"
      role="alert"
      style="display: none;"
    >
      <div
        class="v-alert__wrapper"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"
        />
        <div
          class="v-alert__content"
        >
          
    
  
        </div>
        <button
          aria-label="Close"
          class="v-alert__dismissible v-btn v-btn--icon v-btn--round theme--dark v-size--small"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-close-circle theme--dark"
            />
          </span>
        </button>
      </div>
    </div>
  </transition-stub>
   
  <div
    class="report-page__form"
  >
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-text-field__slot"
          >
            <label
              class="v-label theme--light"
              for="input-9"
              style="left: 0px; position: absolute;"
            >
              E-mail
            </label>
            <input
              id="input-9"
              type="text"
            />
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-12"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-12"
              style="left: 0px; position: absolute;"
            >
              Cidade de origem
            </label>
            <input
              autocomplete="off"
              id="input-12"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-17"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-17"
              style="left: 0px; position: absolute;"
            >
              Cidade de destino
            </label>
            <input
              autocomplete="off"
              id="input-17"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-22"
          class="v-input__slot"
          role="button"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-22"
              style="left: 0px; position: absolute;"
            >
              Cid
            </label>
            <div
              class="v-select__selections"
            >
              <input
                aria-readonly="false"
                autocomplete="off"
                id="input-22"
                readonly="readonly"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <button
      class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light elevation-2 v-size--large primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
        enviar
      
      </span>
    </button>
  </div>
</div>
`;

exports[`Pages / ReportPage when user clicks in submit should render updated page 1`] = `
<div
  class="report-page"
>
  <transition-stub
    class="report-page__alert"
    name="scale-transition"
  >
    <div
      class="v-alert v-sheet theme--dark success"
      role="alert"
      style=""
    >
      <div
        class="v-alert__wrapper"
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate v-alert__icon mdi mdi-check-circle theme--dark"
        />
        <div
          class="v-alert__content"
        >
          
    Solicitação enviada com sucesso!
  
        </div>
        <button
          aria-label="Close"
          class="v-alert__dismissible v-btn v-btn--icon v-btn--round theme--dark v-size--small"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-close-circle theme--dark"
            />
          </span>
        </button>
      </div>
    </div>
  </transition-stub>
   
  <div
    class="report-page__form"
  >
    <a
      class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg v-btn--router theme--light elevation-2 v-size--large primary"
    >
      <span
        class="v-btn__content"
      >
        
      voltar para home
    
      </span>
    </a>
  </div>
</div>
`;
